### PR TITLE
Publish to DockerHub through GitHub Actions

### LINKED ISSUE #10: Please provide version tags for tonistiigi/xx:golang
I've been using successfully tonistiigi/xx:golang for compiling my golang apps to different architectures including linux/386 (with CGO_ENABLED=0) because the image was not a multi-arch one. The last one is targeting different architectures but linux/386 is not among them and my build started to fail. Unfortunately, I can't pick the previous (non-manifest v2) image because this image doesn't contain a version in its tag. 
Since this image is widely popular and used in multiple builds would it be possible to add version tags or maybe produce an arch-agnostic image that supports builds with CGO_ENABLED=0?

### PR DESCRIPTION
**Pull request by crazy-max**
Closes #10 

Add `publish` job to push image to DockerHub.

* On `push tag` like `v1.2.3` will create `golang-1.2.3`, `golang` docker tags and push to DockerHub
* On `push` will create `golang-<branch>` docker tag and push to DockerHub
* On `pull_request` will only build locally

@tonistiigi You will need to create secrets `DOCKERHUB_USERNAME` and `DOCKERHUB_TOKEN`.

### CODE REVIEW

File: .github/workflows/golang.yml [hunk 1]
```diff
@@ -2,10 +2,16 @@ name: golang
 
 on:
   push:
+    branches:
+      - 'master'
+    tags:
+      - 'v*'
```
Comment (tonistiigi) [inline .github/workflows/golang.yml:8]: guess this should be `golang/v*`

File: .github/workflows/golang.yml [hunk 2]
```diff
@@ -31,3 +37,44 @@ jobs:
...
+        name: Prepare
+        id: prep
+        run: |
+          DOCKER_IMAGE=tonistiigi/xx
+          VERSION=golang-edge
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/v}
+          fi
+          TAGS="${DOCKER_IMAGE}:golang-${VERSION}"
+          if [[ $VERSION =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
```
Comment (tonistiigi) [inline .github/workflows/golang.yml:57]: assuming tag is `golang/v*` we can just replace `/` with `-` and add constant `tonistiigi/xx:golang` as the latest version.

I prefer `golang-master` instead `golang-edge`. `golang-<branch>`

File: .github/workflows/golang.yml [hunk 3]
```diff
@@ -31,3 +37,44 @@ jobs:
...
           platforms: ${{ steps.buildx.outputs.platforms }}
+
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Prepare
```
Comment (tonistiigi) [inline .github/workflows/golang.yml:48]: "Evaluate target names"

**Review by tonistiigi — COMMENTED** [review]
This makes me also think of defining bake target instead but can try that out later.

File: .github/workflows/golang.yml (hunk 2 above)
Comment (crazy-max) [reply .github/workflows/golang.yml:59]: Done and updated [my comment](https://github.com/tonistiigi/xx/pull/11#issue-486055664) to reflect that.

File: .github/workflows/golang.yml [hunk 4]
```diff
@@ -31,3 +37,46 @@ jobs:
...
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Evaluate target names
+        id: prep
+        run: |
+          DOCKER_IMAGE=tonistiigi/xx
+          VERSION=dummy
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
```
Comment (tonistiigi) [inline .github/workflows/golang.yml:53]: `refs/tags/golang/v*` here for safety?

Comment (crazy-max) [reply .github/workflows/golang.yml]: Should not happen but who knows :) Done

**Review by tonistiigi — APPROVED** [review]

**Comment by tonistiigi** [comment]
Thanks

**Pull request merged**